### PR TITLE
Log customization doc: IncludeScopes

### PR DESCRIPTION
**Pull request by TimothyMothra**
#2999

## Changes
- section for `OpenTelemetryLoggerOptions.IncludeScopes`
- code example

Please provide a brief description of the changes here.

For significant contributions please make sure you have completed the following items:

* [ ] Appropriate `CHANGELOG.md` updated for non-trivial changes
* [ ] Design discussion issue #
* [ ] Changes in public API reviewed

### CODE REVIEW

File: docs/logs/customizing-the-sdk/Program.cs [hunk 1]
```diff
@@ -41,5 +42,12 @@ public static void Main()
         logger.LogInformation("Hello Information");
         logger.LogWarning("Hello Warning");
         logger.LogError("Hello Error");
+
+        // DEMO of IncludeScopes
```
Comment (reyang) [inline docs/logs/customizing-the-sdk/Program.cs:46]: Need more realistic example: https://github.com/open-telemetry/opentelemetry-dotnet/blob/9da4ff798ade9172bb4e14dacc6a847b72531975/docs/logs/extending-the-sdk/Program.cs#L62-L66

File: docs/logs/customizing-the-sdk/README.md [hunk 2]
```diff
@@ -14,7 +14,13 @@ TODO
 
 ### IncludeScopes
 
-TODO
+A "scope" is an ILogger concept that can group a set of logical operations and
```
Comment (reyang) [inline docs/logs/customizing-the-sdk/README.md:17]: Link to some official doc.

Comment (TimothyMothra) [reply docs/logs/customizing-the-sdk/README.md]: done

File: docs/logs/customizing-the-sdk/Program.cs (hunk 1 above)
Comment (TimothyMothra) [reply docs/logs/customizing-the-sdk/Program.cs]: copied the demo from extending-the-sdk

**Review by reyang — APPROVED** [review]

**Review by Kielek — APPROVED** [review]

File: docs/logs/customizing-the-sdk/README.md [hunk 3]
```diff
@@ -14,7 +14,14 @@ TODO
 
 ### IncludeScopes
 
-TODO
+A "[scope](https://docs.microsoft.com/dotnet/core/extensions/logging#log-scopes)"
+is an ILogger concept that can group a set of logical operations and attach
```
Comment (pellared) [inline docs/logs/customizing-the-sdk/README.md:18]: nit

```suggestion
is an `ILogger` concept that can group a set of logical operations and attach
```

**Review by pellared — APPROVED** [review]

File: docs/logs/customizing-the-sdk/Program.cs [hunk 4]
```diff
@@ -41,5 +42,12 @@ public static void Main()
         logger.LogInformation("Hello Information");
         logger.LogWarning("Hello Warning");
         logger.LogError("Hello Error");
+
+        // log with scopes
+        using (logger.BeginScope("[operation]"))
```
Comment (cijothomas) [inline docs/logs/customizing-the-sdk/Program.cs:47]:
```suggestion
        using (logger.BeginScope("operation"))
```

File: docs/logs/customizing-the-sdk/README.md [hunk 5]
```diff
@@ -14,7 +14,14 @@ TODO
...
+data to each log created as part of a set.
+
+`IncludeScopes` is off by default. Setting this to `true` will include all
+scopes with the exported `LogRecord`.
```
Comment (cijothomas) [inline docs/logs/customizing-the-sdk/README.md:22]: Add a note saying that - Consult the individual `Exporter` docs to know more about how each Exporters process scope information.

**Review by cijothomas — APPROVED** [review]

**Review by cijothomas — APPROVED** [review]

**Pull request merged**